slo-workload/src/Commands/RunCommand.php

```diff
@@ -0,0 +1,421 @@
+        $msgQueue = msg_get_queue($queueId);
+
+        $registry->wipeStorage();
+        $pushGateway->push($registry, 'workload-php', [
+            'sdk' => 'php',
+            'sdkVersion' => Ydb::VERSION
+        ]);
+        $pushGateway->delete('workload-php', [
+            'sdk' => 'php',
+            'sdkVersion' => Ydb::VERSION
+        ]);
+
+        $lastPushTime = microtime(true);
+
+        foreach ($this->errors as $error) {
+            $errors->incBy(0, ['read', $error, 'finally']);
+            $errors->incBy(0, ['write', $error, 'finally']);
+            $errors->incBy(0, ['read', $error, 'retried']);
+            $errors->incBy(0, ['write', $error, 'retried']);
+        }
+
+        $pushGateway->push($registry, "workload-php", [
+            'sdk' => 'php',
+            'sdkVersion' => Ydb::VERSION
+        ]);
+
+
+        while (microtime(true) <= $startTime + $time) {
+            while (msg_receive($msgQueue, 1, $msgType, 1024, $message)) {
+                switch ($message['type']) {
+                    case 'reset':
+                        $pushGateway->delete('workload-php', [
+                            'sdk' => 'php',
+                            'sdkVersion' => Ydb::VERSION
+                        ]);
+                        return;
+                    case  'start':
+                        $inflight->inc([$message['job']]);
+                        break;
+                    case 'ok':
+                        $inflight->dec([$message['job']]);
+                        $latencies->observe($message['latency'], [$message['job'], $message['type']]);
+                        $attempts->observe($message['attempts'], [$message['job'], $message['type']]);
+                        $oks->inc([$message['job']]);
+                        break;
+                    case 'err':
+                        $inflight->dec([$message['job']]);
+                        $latencies->observe($message['latency'], [$message['job'], $message['type']]);
+                        $attempts->observe($message['attempts'], [$message['job'], $message['type']]);
+                        $notOks->inc([$message['job']]);
+                        $errors->inc([$message['job'], $message['error'], 'finally']);
+                        break;
+                    case 'retried':
+                        $errors->inc([$message['job'], $message['error'], 'retried']);
+                        break;
+                }
+
+                if ((microtime(true) - $lastPushTime) * 1000 > $reportPeriod) {
+                    $pushGateway->push($registry, "workload-php", [
+                        'sdk' => 'php',
+                        'sdkVersion' => Ydb::VERSION
+                    ]);
+                    $lastPushTime = microtime(true);
+                }
+                if (microtime(true) + 0.01 >= $startTime + $time) {
```

Comment:
better to use one final code block.

it may be extracted as function if no comfort way - for wipe registry, the function may be used at start of metrics too.